cli/tests/containerize.bats

```diff
@@ -56,94 +57,80 @@ podman_home_setup() {
+  export XDG_RUNTIME_DIR="$FLOX_TEST_HOME/run"
+  export XDG_CONFIG_HOME="$FLOX_TEST_HOME/.config"
+
+  mkdir -p "$XDG_DATA_HOME"
+  mkdir -p "$XDG_STATE_HOME"
+  mkdir -p "$XDG_CACHE_HOME"
+  mkdir -p "$XDG_RUNTIME_DIR"
+  mkdir -p "$XDG_CONFIG_HOME"
+
+  # Set the flox-specific directories to point to this home-tempdir
+  export FLOX_CACHE_DIR="$XDG_CACHE_HOME/flox"
   export FLOX_CONFIG_DIR="$XDG_CONFIG_HOME/flox"
   export FLOX_DATA_HOME="$XDG_DATA_HOME/flox"
   export FLOX_STATE_HOME="$XDG_STATE_HOME/flox"
   export FLOX_META="$FLOX_CACHE_DIR/meta"
   export FLOX_ENVIRONMENTS="$FLOX_DATA_HOME/environments"
-  export HOME="${FLOX_TEST_HOME:-$HOME}"
-}
-
-
-podman_dirs_setup() {
-  # Podman creates deeply nested directories and stores sockets in some of them,
-  # so we need to create locations to store those with shorter paths than what
-  # we'd get nesting them under `/tmp/nix-shell.XXXXXX/bats-run-XXXXXX`.
-  export SHORT_TMP="$(mktemp -d "/tmp/XXXXXX")"
-  export TMPDIR="$SHORT_TMP"
-  export XDG_CONFIG_HOME="$SHORT_TMP/.config"
-  export XDG_DATA_HOME="$SHORT_TMP/.local/share"
-  export XDG_RUNTIME_DIR="$SHORT_TMP/run"
-  echo "Podman XDG root: $SHORT_TMP" >&3
-  mkdir -p "$XDG_CONFIG_HOME/containers"
-  mkdir -p "$XDG_DATA_HOME"
-  mkdir -p "$XDG_RUNTIME_DIR"
 
-  # Don't require Rosetta unless cross-compiling on builders.
-  if [[ "${NIX_SYSTEM}" != "x86_64-darwin" ]]; then
-    cat > "$XDG_CONFIG_HOME/containers/containers.conf" <<'EOF'
-[machine]
-rosetta = false
-EOF
-  fi
-  
-  podman_home_setup
-  podman_xdg_vars_setup "$FLOX_TEST_HOME" "$SHORT_TMP"
-  podman_flox_vars_setup
+  # Set HOME
+  export HOME="${FLOX_TEST_HOME:?FLOX_TEST_HOME was unset or null}"
 }
 
 start_podman_machine() {
-  machine="$(podman machine list -n)"
-  if [ -z "$machine" ]; then
+  # For some of these calls you need to not only close FD 3, but also FD 4.
+  # I have no idea why.
+
+  # This output should be "flox-containerize-vm*" or "" if the machine doesn't exist yet
+  machines="$(XDG_DATA_HOME="$REAL_XDG_DATA_HOME" XDG_CONFIG_HOME="$REAL_XDG_CONFIG_HOME" podman machine list -n -q)"
+  if [ -z "$machines" ]; then
     echo "Creating podman machine" >&3
-    podman machine init -v /tmp:/tmp -v /Users:/Users -v /private:/private
+    # These variables are the same ones set on the wrapper script in `setup()`
+    XDG_DATA_HOME="$REAL_XDG_DATA_HOME" XDG_CONFIG_HOME="$REAL_XDG_CONFIG_HOME" podman machine init -v /tmp:/tmp -v /Users:/Users -v /private:/private flox-containerize-vm 3>&- 4>&-
   fi
   echo "Starting podman machine" >&3
-  podman machine start
+  # These variables are the same ones set on the wrapper script in `setup()`.
+  XDG_DATA_HOME="$REAL_XDG_DATA_HOME" XDG_CONFIG_HOME="$REAL_XDG_CONFIG_HOME" podman machine start flox-containerize-vm 3>&- 4>&-
 }
 
 # ---------------------------------------------------------------------------- #
 
+# Identical to `setup_isolated_flox` but doesn't handle FLOX_CACHE_DIR because
```

Comment:
Shouldn’t this already go into the “per test” test does?

Reply:
Unclear what you mean...